[Macros] Don't allow types to shadow freestanding macros.

DougGregor opened the pull request:
When performing name lookup for freestanding macros (e.g., after the `#`), don't allow types to shadow macros from imported libraries.

Fixes rdar://110429368.

DougGregor committed:
[Macros] Don't allow types to shadow freestanding macros.

When performing name lookup for freestanding macros (e.g., after the
`#`), don't allow types to shadow macros from imported libraries.

Fixes rdar://110429368.

DougGregor commented:
@swift-ci please smoke test

bnbarham commented on test/Macros/macro_shadowing.swift:
```diff
@@ -0,0 +1,20 @@
+struct stringify<T> { }
+
+func testStringify(a: Int, b: Int) {
+  _ = #stringify(a + b)
```
What's the diagnostic now if there's a type but no macro? I assume just a "cannot find macro..."?

DougGregor replied:
Yes, the same as it was before, because everything that isn't a macro gets filtered out.

DougGregor committed:
Remove `findMacroForCustomAttr` in favor of `namelookup::lookupMacros`

The former was effectively a copy of the latter.

DougGregor committed:
[Macros] Fix odd test case

The implementation was previously finding the wrong macro declaration,
leading to incorrect results recorded as the expected results.

DougGregor commented:
@swift-ci please smoke test
